internal/cnpi/plugin/mapping.go

```diff
@@ -22,15 +22,16 @@ import (
 	"github.com/cloudnative-pg/cnpg-i/pkg/lifecycle"
 )
 
-// The OperationVerb corresponds to the Kubernetes API method
+// The OperationVerb corresponds to the CNPG-I lifecycle operation verb
 type OperationVerb string
 
-// A Kubernetes operation verb
+// A lifecycle operation verb
 const (
-	OperationVerbPatch  OperationVerb = "PATCH"
-	OperationVerbUpdate OperationVerb = "UPDATE"
-	OperationVerbCreate OperationVerb = "CREATE"
-	OperationVerbDelete OperationVerb = "DELETE"
+	OperationVerbPatch    OperationVerb = "PATCH"
+	OperationVerbUpdate   OperationVerb = "UPDATE"
+	OperationVerbCreate   OperationVerb = "CREATE"
+	OperationVerbDelete   OperationVerb = "DELETE"
+	OperationVerbEvaluate OperationVerb = "EVALUATE"
```

Comment:
In my opinion, Evaluate as a verb would give the impression of having no side-effects.
As this seems to be a mutation applied on the PodSpec, before actually creating the pod, I might suggest other verbs.
BTW made the same comments. Attaching here so they show up in reviews.

Reply:
I agree with you on the `Evaluate` concept. In the current implementation it has no side effects as the result is only used to evaluate whether the Pod definition is different than the one that would be generated.
The `Create` verb is the one actually used to create the new Pod.

Naming is difficult... at this moment I don't perceive `Evaluate` as misleading.